### DISCUSSION TITLE
[Evercam] Programatically attach panels on the left side for all users

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#2856] [Q&A] [answered]
Hello,

In our current application we are trying to change the ITwin widget layout automatically for all users so for example instead of all the widgets being on the right side like this: 
![Screenshot_1](https://user-images.githubusercontent.com/29980658/145058856-498eddb1-e8d0-42d5-abf6-7c9c46b342de.jpg)

To all the widgets being on the left side like this :
![Screenshot_2](https://user-images.githubusercontent.com/29980658/145058968-7adfb9e6-4bd4-4616-95bf-ab0a86155bdd.jpg)

**Comment by aruniverse** [comment, marked as answer]
In your UiItemsProvider.provideWidget, set [StagePanelLocation](https://www.itwinjs.org/reference/ui-abstract/widget/stagepanellocation/) to `Left`.


```
 public provideWidgets(
    _stageId: string,
    stageUsage: string,
    location: StagePanelLocation,
    section?: StagePanelSection | undefined
  ): ReadonlyArray<AbstractWidgetProps> {
    const widgets: AbstractWidgetProps[] = [];
    if (
      stageUsage === StageUsage.General &&
      location === StagePanelLocation.Left&&
      section === StagePanelSection.End
    ) {
      widgets.push({...});
    }

    return widgets;
  }
```

**Reply by salah-walid** [reply]
Thanks, is there a way to change the default widgets like the three view as well

**Reply by aruniverse** [reply]
Each user can move the default widgets over manually (drag and drop to the left or any location they want). 

Unfortunately, there is no easy way to override the location of the default widgets for all users at the moment.
You'll need to:
1. `hideTreeView` and `hidePropertyGrid`, see the [defaultUiConfig prop](https://github.com/itwin/viewer/tree/master/packages/modules/web-viewer-react#optional)
2. Write a new UiItemsProvider that places the widgets in the left stage panel
	- Take a look at [TreeWidget](https://github.com/iTwin/viewer-components-react/blob/master/packages/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx) and [PropertyGrid](https://github.com/iTwin/viewer-components-react/blob/master/packages/property-grid/src/PropertyGridUiItemsProvider.tsx)
3. Register your new UiItemsProviders
